### PR TITLE
watch lsp state instead

### LINKED ISSUE #693: Wrong receive limit in the UI
image

### PR DESCRIPTION
**Pull request by ubbabeck**
Fixes #693

### CODE REVIEW

File: lib/widgets/receivable_btc_box.dart [hunk 1]
```diff
@@ -57,11 +57,9 @@ class ReceivableBTCBoxState extends State<ReceivableBTCBox> {
...
                     widget.receiveLabel ??
                         texts.invoice_receive_label(
-                          currencyState.bitcoinCurrency.format(
-                            (isChannelOpeningAvailable)
-                                ? accountState.maxInboundLiquidity
-                                : accountState.maxAllowedToReceive,
-                          ),
+                          currencyState.bitcoinCurrency.format((!isChannelOpeningAvailable)
+                              ? accountState.maxInboundLiquidity
+                              : accountState.maxAllowedToReceive),
```
Comment (erdemyerebasmaz) [inline lib/widgets/receivable_btc_box.dart:62]: NIT Suggestion:

Swap conditions for readability.
```suggestion
                          currencyState.bitcoinCurrency.format((isChannelOpeningAvailable)
                              ? accountState.maxAllowedToReceive
                              : accountState.maxInboundLiquidity),
```

File: lib/widgets/receivable_btc_box.dart [hunk 2]
```diff
@@ -30,10 +30,10 @@ class ReceivableBTCBoxState extends State<ReceivableBTCBox> {
   late final texts = context.texts();
   late final currencyState = context.read<CurrencyBloc>().state;
   late final accountState = context.read<AccountBloc>().state;
-  late final lspState = context.read<LSPBloc>().state;
 
   @override
   Widget build(BuildContext context) {
+    final lspState = context.watch<LSPBloc>().state;
```
Comment (erdemyerebasmaz) [inline lib/widgets/receivable_btc_box.dart:36]: NIT Suggestion:

Think there's nothing stopping us from defining these in `build`:

```
    final texts = context.texts();
    final currencyState = context.read<CurrencyBloc>().state;
    final accountState = context.read<AccountBloc>().state;
    final lspState = context.read<LSPBloc>().state;
```

Please check logs & verify the UI has no issues after this change.

**Review by erdemyerebasmaz — CHANGES_REQUESTED** [review]
LGTM

2 nit suggestions

File: lib/widgets/receivable_btc_box.dart (hunk 1 above)
Comment (ubbabeck) [reply lib/widgets/receivable_btc_box.dart]: Done

File: lib/widgets/receivable_btc_box.dart (hunk 2 above)
Comment (ubbabeck) [reply lib/widgets/receivable_btc_box.dart]: Added

**Review by erdemyerebasmaz — APPROVED** [review]

**Pull request merged**